src/value.zig

```diff
@@ -1233,15 +1233,161 @@ pub const Value = extern union {
     }
 
     pub fn eql(a: Value, b: Value) bool {
-        if (a.tag() == b.tag() and a.tag() == .enum_literal) {
-            const a_name = @fieldParentPtr(Payload.Bytes, "base", a.ptr_otherwise).data;
-            const b_name = @fieldParentPtr(Payload.Bytes, "base", b.ptr_otherwise).data;
-            return std.mem.eql(u8, a_name, b_name);
+        if (a.tag() == b.tag()) {
+            if (a.tag() == .void_value or a.tag() == .null_value) {
+                return true;
+            } else if (a.tag() == .enum_literal) {
+                const a_name = @fieldParentPtr(Payload.Bytes, "base", a.ptr_otherwise).data;
+                const b_name = @fieldParentPtr(Payload.Bytes, "base", b.ptr_otherwise).data;
+                return std.mem.eql(u8, a_name, b_name);
+            }
+        }
+        if (a.isType() and b.isType()) {
+            // 128 bytes should be enough to hold both types
```

Comment:
hmm would this crash the compiler then?

```zig
test "" {
    const T1 = u8;
    const T2 = *********************************************************************************************************************************u8;
    const eql = T1 == T2;
}
```

Reply:
No, `toType` only allocates for `.int_type` and `.error_set` with `.error_set` requiring 16 bytes.